server/auth/auth.go

```diff
@@ -29,18 +29,19 @@ func unixMsNow() time.Time {
 // Storage updates and fetches account-related data from what is presumably a
 // database.
 type Storage interface {
-	// CloseAccount closes the account for violation of the specified rule.
-	CloseAccount(account.AccountID, account.Rule) error
-	// RestoreAccount opens an account that was closed by CloseAccount.
-	RestoreAccount(account.AccountID) error
 	// Account retrieves account info for the specified account ID.
-	Account(account.AccountID) (acct *account.Account, paid, open bool)
+	Account(account.AccountID) (acct *account.Account, paid bool)
 	CompletedUserOrders(aid account.AccountID, N int) (oids []order.OrderID, compTimes []int64, err error)
 	ExecutedCancelsForUser(aid account.AccountID, N int) (oids, targets []order.OrderID, execTimes []int64, err error)
 	AllActiveUserMatches(aid account.AccountID) ([]*db.MatchData, error)
 	CreateAccount(*account.Account) (string, error)
 	AccountRegAddr(account.AccountID) (string, error)
 	PayAccount(account.AccountID, []byte) error
+	InsertPenalty(penalty *db.Penalty) (id int64)
+	ForgivePenalty(id int64) error
+	ForgivePenalties(aid account.AccountID) error
+	Penalties(aid account.AccountID) (penalties []*db.Penalty, err error)
+	AllPenalties(aid account.AccountID) (penalties []*db.Penalty, err error)
```

Comment:
These can be merged
```suggestion
	Penalties(aid account.AccountID, active bool) (penalties []*db.Penalty, err error)
```
where `active` means not forgiven and not expired.